magicauth/send_token.py

```diff
@@ -37,10 +37,15 @@ def send_email(self, user, user_email, token, extra_context=None):
         html_template = magicauth_settings.EMAIL_HTML_TEMPLATE
         text_template = magicauth_settings.EMAIL_TEXT_TEMPLATE
         from_email = magicauth_settings.FROM_EMAIL
+        request_site = self.request.get_host()
```

Comment:
get_host will already check against ALLOWED_HOSTS (and not break with "*").

https://docs.djangoproject.com/en/3.1/ref/settings/#allowed-hosts says : 
`If the Host header (or X-Forwarded-Host if USE_X_FORWARDED_HOST is enabled) does not match any value in this list, the django.http.HttpRequest.get_host() method will raise SuspiciousOperation.`

So I think the most appropriate is to 
 - remove the homemade check against ALLOWED_HOSTS
 - add a comment mentioning that get_host checks against ALLOWED_HOSTS (so that this whole research is not lost)
 - test it!